demo/full/scripts/components/VideoThumbnail.jsx

```diff
@@ -0,0 +1,155 @@
+        "opacity": "50%",
+        "display": "flex",
+        "justify-content": "center",
+        "align-items": "center",
+      },
+      spinnerStyle: {
+        "width": "50%",
+        "margin": "auto",
+      },
+      displaySpinner : true,
+    };
+    this.lastSetTime = undefined;
+    this._videoElement = undefined;
+  }
+
+  correctImagePosition() {
+    if (this.positionIsCorrected) {
+      return;
+    }
+    const { xPosition } = this.props;
+
+    if (isNaN(+xPosition) || !this.element) {
+      return null;
+    }
+
+    const style = {
+      transform: `translate(${xPosition}px, -136px)`,
+    };
+
+    this.positionIsCorrected = true;
+    this.setState({ style });
+  }
+
+  showSpinner() {
+    if (this.state.displaySpinner !== true) {
+      this.setState({ displaySpinner: true });
+    }
+  }
+
+  hideSpinner() {
+    if (this.state.displaySpinner !== false) {
+      this.setState({ displaySpinner: false });
+    }
+  }
+
+  componentWillReceiveProps() {
+    this.positionIsCorrected = false;
+  }
+
+  componentDidMount() {
+    this.correctImagePosition();
+    if (this._videoElement !== undefined) {
+      this.props.player.dispatch("ATTACH_VIDEO_THUMBNAIL_LOADER", this._videoElement);
+    }
+  }
+
+  componentDidUpdate() {
+    this.correctImagePosition();
+  }
+
+  componentWillUnmount() {
+    const { player, attachedVideoThumbnailLoader } = this.props;
+    const videoThumbnailLoader = attachedVideoThumbnailLoader;
+    if (videoThumbnailLoader) {
+      videoThumbnailLoader.dispose();
+      player.dispatch("REMOVE_VIDEO_THUMBNAIL_LOADER");
+    }
+    this._videoElement = undefined;
+  }
+
+  render() {
+    const { style, divSpinnerStyle, spinnerStyle } = this.state;
+
+    const videoThumbnailLoader = this.props.attachedVideoThumbnailLoader;
+    if (videoThumbnailLoader) {
+      const { time } = this.props;
+      const roundedTime = Math.round(time);
+      let spinnerTimeout;
+      // Only show spinner when time has changed
+      if (this.lastSetTime !== roundedTime) {
+        // Wait a little before displaying spinner, to
+        // be sure loading takes time
+        spinnerTimeout = setTimeout(() => {
+          this.showSpinner();
+        }, 300);
+      }
+      this.lastSetTime = roundedTime;
+      videoThumbnailLoader.setTime(roundedTime)
+        .then(() => {
+          clearTimeout(spinnerTimeout);
+          this.hideSpinner();
+        })
+        .catch(() => {
+          clearTimeout(spinnerTimeout);
+          this.hideSpinner();
+        });
+    }
+
+    const divToDisplay = <div
+      className="thumbnail-wrapper"
+      style={style}
+      ref={el => this.element = el}
+    >
+      {
+        this.state.displaySpinner ?
+          <div style={divSpinnerStyle}>
+            <img
+              src="../assets/spinner.gif"
+              style={spinnerStyle}
+            />
+          </div> :
+          null
+      }
+      <video ref={(videoElement) => {
```

Comment:
We can use `React.createRef()`, using ref like this is the old way to do this, and not recommended anymore:
- https://fr.reactjs.org/docs/refs-and-the-dom.html#creating-refs

Reply:
What does `myRef.current` is equal to when not set yet? `null`/`undefined`? The documentation you linked to is not really clear about this

Reply:
Ah! `null`, found it later

Reply:
Done. I first merged #931 into this PR because it changes the demo code a lot.

There's still a UNSAFE_componentWillReceiveProps in there, about I did not take the time to think/read about the most idiomatic way to write the corresponding logic. Plus we may probably rewrite that code with hooks in the future.
Nothing should be unsafe about it though.